src/main/java/io/confluent/connect/elasticsearch/AsyncOffsetTracker.java

```diff
@@ -93,11 +96,20 @@ public synchronized void closePartitions(Collection<TopicPartition> topicPartiti
    * Older records can be re-added, and the same Offset object will be return if its
    * offset hasn't been reported yet.
    * @param sinkRecord record to add
+   * @param assignment set of topic partitions, assigned to this task
    * @return offset state record that can be used to mark the record as processed
    */
-  public synchronized OffsetState addPendingRecord(SinkRecord sinkRecord) {
+  public synchronized OffsetState addPendingRecord(
+      SinkRecord sinkRecord,
+      Set<TopicPartition> assignment
```

Comment:
This looks like an implementation specific detail, not something that should be part of the interface. 
I'd move it to the constructor, probably receiving a supplier of the assignment.